tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1155,6 +1155,31 @@ def topic_type(t, topic_types):
         topics.sort()
         print('\n'.join(["%s%s"%(indent, t) for t in topics]))
 
+def get_topic_list(publishers=True, subscribers=True):
+    def topic_type(t, topic_types):
+        matches = [t_type for t_name, t_type in topic_types if t_name == t]
+        if matches:
+            return matches[0]
+        return 'unknown type'
+
+    # Return an array of tuples; (<topic>, <type>, <node_count>)
+    master = rosgraph.Master('/rostopic')
+    state = master.getSystemState()
+    topic_types = _master_get_topic_types(master)
+    
+    pubs, subs, _ = state
+    publishers = []
```

Comment:
This line is overwriting the function parameter.

Same below for `subscribers`.